### ISSUE TITLE
waitForElementToDisappear

### ISSUE THREAD

**Issue by mastertinner** [kentcdodds/dom-testing-library#206]
### Describe the feature you'd like:

I love the `waitForElement` function. However, there are cases where I don't add an element but rather remove existing ones. Is there a similar `waitForElementToDisappear` function or is there an easy way to achieve that using the existing functions?

### Suggested implementation:

```javascript
await waitForElementToDisappear(() => getByText('my button'))
...
```

Currently I'm using `process.nextTick` but that seems like a hack to me:

```
process.nextTick(() => {
    ...
    done()
})
```

**Comment by kentcdodds** [comment]
Forgot about that! Done :)

Now let's discuss.

So you can do this today with:

```js
await wait(() => expect(getByText('my button')).not.toBeInTheDocument())
```

But that's kinda awkward and having a `waitForElementToBeRemoved` or something similar would be useful. Just not sure of the best name.

**Comment by mastertinner** [comment]
In the meantime, I found out that `waitForDomChange` actually covers most of my use cases. For explicitly wait for the removal of an element, I think the name you suggested (`waitForElementToBeRemoved`) is quite a good one. It's a little verbose, but I think that's worth having the name be explicit about what it does.

**Comment by kentcdodds** [comment]
I'm into it. I say let's do it.

**Comment by lgandecki** [comment]
@kentcdodds I think it would have to be queryByText? Otherwise the getByText itself will fail when the element goes away, and the expectation won't be checked.
Also, I wonder if the awkwardness here is something you feel very strongly? I don't personally see a problem with that wait (that's +/- how I've seen this getting done in "selenium era"). Maybe a documentation change would be a way to go instead of increasing the API surface?

**Comment by kentcdodds** [comment]
No worries @lgandecki, thanks for providing your perspective. You make good points. I wonder though if there could be several document changes before that thing is actually removed. Like a progress bar for example.

It's true that this is really not a very typical thing and the workaround isn't that challenging (you're right, would need to be a `queryBy*`). Hmmm...

**Comment by gabriel403** [comment]
@kentcdodds @lgandecki I stumbled upon this issue and myself today, I need to check a progress loader has finished, as the main list element to check is always in the dom, I could check individual items within the list element but wanted to give this a try and then use snapshots for asserting that the items are there.
I tried 
```
      await wait(() => expect(getByTestId('progress-loader')).not.toBeInTheDocument());
```
but as stated that fails as the getByTestId fails when the loader is done.
I also tried 
```
      await wait(() => expect(queryByTestId('progress-loader')).not.toBeInTheDocument());
```
but now I get `TypeError: expect(...).not.toBeInTheDocument is not a function`. If you have any advice for this I'd appreciate it
